### ISSUE TITLE
XcodeGen 2.1.0 fails to build as an SPM dependency

### ISSUE THREAD

**Issue by ksulliva** [yonaskolb/XcodeGen#481]
When adding XcodeGen 2.1.0 as a SPM dependency, compilation fails with the following message:

```
.../XcodeGen.git-2328129159905136123/Sources/ProjectSpec/Linkage.swift:12:9: error: switch must be exhaustive
        switch type {
        ^
.../XcodeGen.git-2328129159905136123/Sources/ProjectSpec/Linkage.swift:12:9: note: add missing case: '.instrumentsPackage'
        switch type {
        ^
```

It looks like the latest release of `xcodeproj` (6.4.0) added another case to `PBXProductType` which isn't being handled.

**Comment by yonaskolb** [comment]
Hmm, this shouldn't happen because there's a `Package.resolved` checked in, which locks xcodeproj to 6.3.0? Perhaps that's only used if building the current package on it's own and not if it's used as a dependency

**Comment by icanswiftabit** [comment]
This is happening again. I'm trying to use `XcodeGenKit` as a dependency. When I'm trying build my project I'm getting 
```
.../SourcePackages/checkouts/XcodeGen/Sources/ProjectSpec/Linkage.swift:12:9: Switch must be exhaustive
```
` .xcFramework:` case is missing.

**Comment by yonaskolb** [comment]
Ok, we’ll have to lock down the XcodeProj version, as it keeps introducing breaking changes.
Using a package as a dependency ignores its package.resolved

**Comment by yonaskolb** [comment]
Fixed in 2.13.0
